framework/include/base/MooseObject.h

```diff
@@ -82,6 +152,9 @@ class MooseObject :
   /// The name of this object, reference to value stored in InputParameters
   const std::string & _name;
 
+  /// Reference to the "enable" InputParaemters, used by Controls for toggling on/off MooseObjects
+  const bool & _enabled;
```

Comment:
We could add an interface class for doing this: "Disablable" :) 

This way only objects that this makes sense for have the parameter.